File: 204 Count Primes/count_primes.java
/*	
 *	LeetCode Submissions by Xinyu Chen
 *	Count Primes
 *	https://leetcode.com/problems/count-primes/
 *	Runtime: 288 ms
 */
 
public class Solution {
    public int countPrimes(int n) {
        if (n < 2) return 0;
        boolean[] numbers = new boolean[n];
        numbers[0] = false;
        numbers[1] = false;
        for (int i = 2; i < n; i++) numbers[i] = true;

        int p = 1;
        while (++p < Math.sqrt(n)) {
            if (numbers[p])
                for (int i = p*p; i < n; i += p) numbers[i] = false;
        }

        int counter = 0;
        for (boolean b : numbers)
            if (b) counter++;
        return counter;
    }
}

File: 204 Count Primes/count_primes.py
# LeetCode Submissions by Xinyu Chen
# Count Primes
# https://leetcode.com/problems/count-primes/
# Runtime: 868 ms

class Solution:
    # @param {integer} n
    # @return {integer}
    def countPrimes(self, n):
        if n < 2:
            return 0
        numbers = [True] * n
        numbers[0] = False
        numbers[1] = False
        p = 2
        while p < math.sqrt(n):
            if numbers[p]:
                for i in range(p*p, n, p):
                    numbers[i] = False
            p += 1
        
        counter = 0
        for i in range(n):
            if numbers[i]:
                counter += 1
        return counter